@kiva/kv-components/tests/unit/specs/components/KvTabs.spec.js

```diff
@@ -96,4 +97,74 @@ describe('KvTabs', () => {
 		userEvent.keyboard('{Home}');
 		expect(await findByText('First Panel')).toBeVisible();
 	});
+
+	it('verifies setTab method when called by a ref', async () => {
+		let selectedIndexCheck = null;
+		const {
+			getByRole,
+			getAllByRole,
+			getByText,
+		} = render({
+			components: {
+				KvTabs,
+				KvTab,
+				KvTabPanel,
+				KvButton,
+			},
+			template: `
+				<div data-testid="tabContainer">
+					<kv-tabs @tab-changed="handleTabChanged" data-testid="tabInstance" ref="tabsRef">
+						<template #tabNav>
+							<kv-tab forPanel="demo-1-first">First</kv-tab>
+							<kv-tab forPanel="demo-1-second">Second</kv-tab>
+							<kv-tab forPanel="demo-1-third">Third</kv-tab>
+							<kv-tab forPanel="demo-1-forth">Fourth is longer</kv-tab>
+						</template>
+						<template #tabPanels>
+							<kv-tab-panel id="demo-1-first"><p>First Panel</p></kv-tab-panel>
+							<kv-tab-panel id="demo-1-second"><p>Second Panel has <br>longer<br>content</p></kv-tab-panel>
+							<kv-tab-panel id="demo-1-third"><p>Third Panel</p></kv-tab-panel>
+							<kv-tab-panel id="demo-1-forth"><p>Fourth Panel</p></kv-tab-panel>
+						</template>
+					</kv-tabs>
+					<kv-button @click.native.prevent="setTabButtonClick(1)" role="tabSetButton">Set Tab</kv-button>
+				</div>
+			`,
+			methods: {
+				handleTabChanged() {
+					this.$nextTick(() => {
+						selectedIndexCheck = this?.$refs?.tabsRef?.tabContext?.selectedIndex;
+					});
+				},
+				setTabButtonClick() {
+					this?.$refs?.tabsRef?.tabContext?.setTab(0);
+				},
+			},
+		}, addVueRouter());
+
+		// check for our elements
+		expect(getByRole('tabSetButton')).toBeDefined();
+		expect(getByRole('tablist')).toBeDefined();
+		expect(getAllByRole('tab').length).toBe(4);
+
+		// Click third tab and expect the third panel to be visible
+		await fireEvent.click(getByText('Third'));
+		expect(getByText('First Panel')).not.toBeVisible();
+		expect(getByText('Third Panel')).toBeVisible();
+		// Test access and updating of property ref
+		// expect the selected index value to be set to our new tab index
+		setTimeout(() => {
+			expect(selectedIndexCheck).toBe(2);
```

Comment:
Because this is in a setTimeout I think it may not be checked before the test completes. It probably wasn't working without this because of nextTick, right? A different way to test this would be to display the selected index in the test component and check for the existence of that using the async findBy* methods:
```
const { ..., findByText } = render({
	...
	data() {
		return { selectedIndex: null };
	},
	template: `
		<div data-testid="tabContainer">
			...
			<p>Tab index is {{ selectedIndex }}</p>
		</div>
	`,
	methods: {
		handleTabChanged() {
			this.$nextTick(() => {
				this.selectedIndex = this?.$refs?.tabsRef?.tabContext?.selectedIndex;
			});
		},
		...
});
...
await fireEvent.click(getByText('Third'));
expect(getByText('First Panel')).not.toBeVisible();
expect(getByText('Third Panel')).toBeVisible();
expect(findByText('Tab index is 2')).toBeVisible();
```

Reply:
Ok I'll test some other options tmrw. They were passing fine without the timeouts in vue 2 and the first one passed in vue3 but not the second, so I just wanted to add a delay.

Reply:
Ok, I was able to drop the next tick as well taking this route! 👍